core/src/main/java/com/graphhopper/routing/DijkstraBidirectionEdgeCHNoSOD.java

```diff
@@ -0,0 +1,221 @@
+        currFrom = createStartEntry(from, weight);
+        pqOpenSetFrom.add(currFrom);
+        if (currTo != null && currTo.adjNode == from) {
+            // special case of identical start and end
+            setFinished();
+            return;
+        }
+        EdgeFilter filter = additionalEdgeFilter;
+        setEdgeFilter(EdgeFilter.ALL_EDGES);
+        fillEdgesFrom();
+        setEdgeFilter(filter);
+
+    }
+
+    @Override
+    public void initTo(int to, double weight) {
+        this.to = to;
+        currTo = createStartEntry(to, weight);
+        pqOpenSetTo.add(currTo);
+        if (currFrom != null && currFrom.adjNode == to) {
+            // special case of identical start and end
+            setFinished();
+            return;
+        }
+        EdgeFilter filter = additionalEdgeFilter;
+        setEdgeFilter(EdgeFilter.ALL_EDGES);
+        fillEdgesTo();
+        setEdgeFilter(filter);
+    }
+
+    private void setFinished() {
+        bestPath.sptEntry = currFrom;
+        bestPath.edgeTo = currTo;
+        bestPath.setWeight(0);
+        finishedFrom = true;
+        finishedTo = true;
+    }
+
+    @Override
+    public Path calcPath(int from, int to) {
+        this.from = from;
+        this.to = to;
+        return super.calcPath(from, to);
+    }
+
+    @Override
+    protected void initCollections(int size) {
+        super.initCollections(Math.min(size, 2000));
+    }
+
+    @Override
+    public boolean finished() {
+        // we need to finish BOTH searches for CH!
+        if (finishedFrom && finishedTo)
+            return true;
+
+        // changed also the final finish condition for CH
+        return currFrom.weight >= bestPath.getWeight() && currTo.weight >= bestPath.getWeight();
+    }
+
+    @Override
+    protected void updateBestPath(EdgeIteratorState edgeState, SPTEntry entryCurrent, int traversalId) {
+        boolean reverse = bestWeightMapFrom == bestWeightMapOther;
+        // special case where the fwd/bwd search runs directly into the opposite node, for example if the highest level
+        // node of the shortest path matches the source or target. in this case one of the searches does not contribute
+        // anything to the shortest path.
+        int oppositeNode = reverse ? from : to;
+        if (edgeState.getAdjNode() == oppositeNode) {
+            if (entryCurrent.weight < bestPath.getWeight()) {
+                bestPath.setSwitchToFrom(reverse);
+                bestPath.setSPTEntry(entryCurrent);
+                bestPath.setWeight(entryCurrent.weight);
+                bestPath.setSPTEntryTo(new SPTEntry(EdgeIterator.NO_EDGE, oppositeNode, 0));
+                return;
+            }
+        }
+        // we can not use in/outEdgeExplorer here because this method will be called within a loop using the 
+        // in/outEdgeExplorer already
+        EdgeExplorer edgeExplorer = graph.createEdgeExplorer(new DefaultEdgeFilter(flagEncoder, reverse, !reverse));
```

Comment:
As creating EdgeExplorers is very expensive we should probably create two additional explorers in the constructor.

Reply:
Sure.